src/meta/src/controller/fragment.rs

```diff
@@ -1120,6 +1121,93 @@ impl CatalogController {
         Ok(chain_fragments)
     }
 
+    /// Find the external stream source info inside the stream node, if any.
+    fn find_stream_source(stream_node: &PbStreamNode) -> Option<&StreamSource> {
+        if let Some(NodeBody::Source(source)) = &stream_node.node_body {
+            if let Some(inner) = &source.source_inner {
+                return Some(inner);
+            }
+        }
+
+        for child in &stream_node.input {
+            if let Some(source) = Self::find_stream_source(child) {
+                return Some(source);
+            }
+        }
+
+        None
+    }
+
+    pub async fn load_source_fragment_ids(
+        &self,
+    ) -> MetaResult<HashMap<SourceId, BTreeSet<FragmentId>>> {
+        let inner = self.inner.read().await;
+        let mut fragments: Vec<(FragmentId, i32, StreamNode)> = Fragment::find()
+            .select_only()
+            .columns([
+                fragment::Column::FragmentId,
+                fragment::Column::FragmentTypeMask,
+                fragment::Column::StreamNode,
+            ])
+            .into_tuple()
+            .all(&inner.db)
+            .await?;
+        fragments.retain(|(_, mask, _)| *mask & PbFragmentTypeFlag::Source as i32 != 0);
```

Comment:
🤔 To be compatible with tables with connector that were created before the new table graph, I think we still need to register the fragment in source manager. Actually even the new dml fragment is registered, it will not be touched as all. There will be no config change worker starts for it.